sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/LogFilteringProcessor.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using OpenTelemetry;
+using OpenTelemetry.Logs;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal class LogFilteringProcessor : BaseProcessor<LogRecord>
+    {
+        private readonly bool _enableSampling;
+        private readonly BaseProcessor<LogRecord> _processor;
+
+        public LogFilteringProcessor(bool enableSampling, BaseProcessor<LogRecord> processor)
+        {
+            _enableSampling = enableSampling;
+            _processor = processor;
+        }
+
+        public override void OnEnd(LogRecord logRecord)
```

Comment:
@cijothomas @rajkumar-rangaraj FYI, let me know if this looks good. I will close the other one.

Reply:
Discussed with Vishwesh the possibility of changing the `AzureMonitorLogExporter` from internal to public in the exporter, and proposed moving all of this implementation to the distro.

Reply:
Synced with @cijothomas as well offline. Here is what I am going to do

1) Make exporter public --> This unblocks exporter users who want to use filtering processor. They will be able to add their own logic.
2) Configure the filtering processor in distro that will sample logs based on ActivityTraceFlags. This will be enabled only via experimental option first and then can be moved to a distro option (public API change)

Will do 1) and 2) as separate PRs.

Reply:
One caveat with the above - It'll allow monitor users to pair any filteringprocessor and achieve custom filtering. but such option won't be available in the distro and distro users will be limited to just filtering based on TraceFlags.

This is probably okay as ILogger will introduce native filtering this year, so that can be leveraged as opposed to custom thing.